pkg/testcase/clusterrestore.go

```diff
@@ -0,0 +1,416 @@
+		product), cluster.ServerIPs[0])
+	Expect(onDemandPathErr).NotTo(HaveOccurred())
+
+	validateS3snapshots(cluster, flags, onDemandPath)
+	shared.LogLevel("info", "successfully validated s3 snapshot save in s3")
+
+	clusterToken, clusterTokenErr := shared.FetchToken(cluster.Config.Product, cluster.ServerIPs[0])
+	Expect(clusterTokenErr).NotTo(HaveOccurred())
+
+	resourceName := os.Getenv("resource_name")
+	ec2, err := aws.AddClient(cluster)
+	Expect(err).NotTo(HaveOccurred(), "error adding aws nodes: %s", err)
+
+	stopInstances(cluster, ec2)
+
+	var serverName []string
+	serverName = append(serverName, fmt.Sprintf("%s-server-fresh", resourceName))
+
+	externalServerIP, _, _, createErr := ec2.CreateInstances(serverName...)
+	Expect(createErr).NotTo(HaveOccurred(), createErr)
+	newServerIP := externalServerIP[0]
+
+	installProduct(cluster, newServerIP, version)
+	shared.LogLevel("info", "%s successfully installed on server: %s", product, newServerIP)
+
+	restoreS3Snapshot(cluster, onDemandPath, clusterToken, newServerIP, flags)
+
+	enableAndStartService(cluster, newServerIP)
+	shared.LogLevel("info", "%s service successfully enabled", product)
+
+	copyCmd := fmt.Sprintf("cp /tmp/%s_kubeconfig /tmp/%s_kubeconfig", resourceName, serverName[0])
+	_, copyCmdErr := shared.RunCommandHost(copyCmd)
+	Expect(copyCmdErr).NotTo(HaveOccurred())
+
+	_, kubeConfigErr := shared.UpdateKubeConfig(newServerIP, serverName[0], product)
+	Expect(kubeConfigErr).NotTo(HaveOccurred())
+
+	postValidationRestore(cluster, newServerIP)
+	shared.PrintClusterState()
+	shared.LogLevel("info", "%s server successfully validated post restore", product)
+}
+
+func cleanVersionData(product, version string) string {
+	var versionClean string
+	versionStr := fmt.Sprintf("%s version ", product)
+	versionCleanUp := strings.TrimPrefix(version, versionStr)
+
+	endChar := strings.Index(versionCleanUp, "(")
+	if endChar != -1 {
+		versionClean = versionCleanUp[:endChar]
+	} else {
+		shared.LogLevel("error", "index returned -1")
+	}
+
+	return versionClean
+}
+
+func setConfigs() {
+	awsConfig = shared.AwsConfig{
+		AccessKeyID:     os.Getenv("AWS_ACCESS_KEY_ID"),
+		SecretAccessKey: os.Getenv("AWS_SECRET_ACCESS_KEY"),
+	}
+}
+
+func takeS3Snapshot(
+	cluster *shared.Cluster,
+	flags *customflag.FlagConfig,
+) {
+	productLocationCmd, findErr := shared.FindPath(cluster.Config.Product, cluster.ServerIPs[0])
+	Expect(findErr).NotTo(HaveOccurred())
+
+	takeSnapshotCmd := fmt.Sprintf("sudo %s etcd-snapshot save --s3 --s3-bucket=%s "+
+		"--s3-folder=%s --s3-region=%s --s3-access-key=%s --s3-secret-key=%s",
+		productLocationCmd, flags.S3Flags.Bucket, flags.S3Flags.Folder, cluster.Aws.Region,
+		awsConfig.AccessKeyID, awsConfig.SecretAccessKey)
+
+	_, takeSnapshotErr := shared.RunCommandOnNode(takeSnapshotCmd, cluster.ServerIPs[0])
+	Expect(takeSnapshotErr).NotTo(HaveOccurred())
+
+	TestDNSAccess(true, false)
```

Comment:
Can you remove this test out of this func and add it on the `TestClusterRestore` func?